mne/channels/layout.py

```diff
@@ -882,13 +888,32 @@ def read_montage(kind, ch_names=None, path=None, scale=True):
         dtype = [('type', 'S8'), ('name', 'S8'),
                  ('x', 'f8'), ('y', 'f8'), ('z', 'f8')]
         data = np.loadtxt(fname, dtype=dtype)
-        pos_ = data[data['type'].astype(np.str) == 'eeg']
-        pos = np.vstack((pos_['x'], pos_['y'], pos_['z'])).T
-        ch_names_ = pos_['name'].astype(np.str)
+        pos = np.vstack((data['x'], data['y'], data['z'])).T
+        ch_names_ = data['name'].astype(np.str)
     else:
         raise ValueError('Currently the "%s" template is not supported.' %
                          kind)
     selection = np.arange(len(pos))
+
+    if transform is True:
+        pos = apply_trans(als_ras_trans_mm, pos)
```

Comment:
> the als to ras transform was in your original read_montage_polhemus function, it was just a transfer.

I know :) I only realized that recently when working with my EEG montages. First I realized that Polhemus does not always output points in ALS, then I was wondering why I still had ended up with correct coordinates, and then I realized that our function for creating the neuromag transform can work with any right handed input coordinate system.  

> to avoid the scaling issue, we could make transform take a string in addition. if it is specified 'polhemus', we do both transforms of rescaling and reorientation.

If we go with a new argument it would be more flexible to have a scale parameter (e.g., `scale=0.001` for input in mm), so anyone using a source different than polhemus could also use the read_montage function, even if the coordinates are in inches :)